core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -1299,39 +1146,38 @@ private ExprValue dayOfWeekToday(Clock clock) {
    * DATE_ADD function implementation for ExprValue.
    *
    * @param functionProperties An FunctionProperties object.
-   * @param datetime ExprValue of Date/Time/Datetime/Timestamp type.
+   * @param datetime ExprValue of Date/Time/Timestamp type.
    * @param interval ExprValue of Interval type, the temporal amount to add.
-   * @return Datetime resulted from `interval` added to `datetime`.
+   * @return Timestamp resulted from `interval` added to `timestamp`.
    */
   private ExprValue exprAddDateInterval(
       FunctionProperties functionProperties, ExprValue datetime, ExprValue interval) {
     return exprDateApplyInterval(functionProperties, datetime, interval.intervalValue(), true);
   }
 
   /**
-   * Adds or subtracts `interval` to/from `datetime`.
+   * Adds or subtracts `interval` to/from `timestamp`.
    *
    * @param functionProperties An FunctionProperties object.
-   * @param datetime A Date/Time/Datetime/Timestamp value to change.
+   * @param datetime A Date/Time/Timestamp value to change.
    * @param interval An Interval to isAdd or subtract.
    * @param isAdd A flag: true to isAdd, false to subtract.
-   * @return Datetime calculated.
+   * @return Timestamp calculated.
    */
   private ExprValue exprDateApplyInterval(
       FunctionProperties functionProperties,
       ExprValue datetime,
       TemporalAmount interval,
       Boolean isAdd) {
-    var dt = extractDateTime(datetime, functionProperties);
-    return new ExprDatetimeValue(isAdd ? dt.plus(interval) : dt.minus(interval));
+    var dt = extractTimestamp(datetime, functionProperties).atZone(UTC_ZONE_ID).toLocalDateTime();
```

Comment:
Why do we need the added `atZone....`?

Reply:
`atZone` is added because timestampValue is stored as a Instant which can't process the MONTH unit (as the days in a month can change) whereas in LocalDateTime it can process MONTH unit as just incrementing the MONTH column. As a general if you see `UTC_ZONE_ID` used it is to process an `instant` to a `localDateTime` as some of the tests store times in the two different data types